javascript/ql/src/semmle/javascript/security/dataflow/CleartextLoggingCustomizations.qll

```diff
@@ -32,6 +32,22 @@ module CleartextLogging {
    * A barrier for clear-text logging of sensitive information.
    */
   abstract class Barrier extends DataFlow::Node { }
+  
+  /**
+   * A call to `.replace()` that seems to mask 
+   */
+  class MaskingReplacer extends Barrier, DataFlow::MethodCallNode {
+  	MaskingReplacer() {
+  	  this.getCalleeName() = "replace" and
+  	  exists(RegExpLiteral reg| 
+  	    reg = this.getArgument(0).getALocalSource().asExpr() and
```

Comment:
you can omit the `this.` prefix.

Reply:
I know, but I often prefer the verbose version.